src/language/walker/blockScopeAwareRuleWalker.ts

```diff
@@ -58,6 +58,16 @@ export abstract class BlockScopeAwareRuleWalker<T, U> extends ScopeAwareRuleWalk
         return;
     }
 
+    public findBlockScope(predicate: (scope: U) => boolean) {
```

Comment:
that creates a copy of the array. I think I made this change based on a discussion about not unnecessarily copying arrays.